dotnet/api/Infrastructure/Data.SqlServer/README.md

```diff
@@ -0,0 +1,17 @@
+# Flattening Migrations
+
+1. Checkout greatest common denominator brach -- typically `master`.
+  a. Master has migrations run on `master` and any additional mainline branches downstream, while development would have migrations that have not yet been run on anything upstream of `development`. Therefore, choose `master`.
+2. Revert your database to the most recent flattened file, and delete all migrations other than the existing flattened migrations.
+3. Create new migration -- this migration will now include everything to get the database to the state it was in after applying every migration you just deleted. Important: This does NOT include any changes manually made to the migration files!
+4. Stash your changes and checkout the branch you'll be working off of (should be a branch off of the least common denominator branch, for example `development`).
+5. Pop the changes from your stash. You should now have your flattened migration for everything that had made it's way to `master`, as well as additional migrations that haven't yet made their way into `master`.
+6. EFCore uses string sorting to apply migrations in order -- thus the timestamp prepended to every migration you create. In this case, it will likely be at or near the end of the list, but we want to move it up to the top of the list to ensure it's run first.
+    * In order to do this, open the flattened migration and ensure it extends `FlattenedMigration` instead of `Migration`.
+    * In the `FlattenedMigration` class, add a new const for the ID of this migration. It should, when sorted as strings, be next in line from the previous flattened migration. Additionally, add it to the static list that contains all the migration ID consts.
+    * Open the `Designer.cs` file of the flattened migration, and update the name in the `[Migration("")]` attribute to use the const you just added.
+    * Rename the files themselves to match what you put in the `[Migration("")]` attribute.
+7. Manually update the flattened migration to call `ValidateFlattenedShouldRun` passing in the ID of the migration you are working off of. This will check if, based on the sort order of the other flattened migrations, this flattened migration has not run on a fresh install. If that method call returns false, you should abort running the flattened migration, as it means this is not a fresh install.
```

Comment:
Can some subset of this comment be added as a comment above the method in that base class?